docs/src/advanced/custom-backend.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Use your own backend
+mainTitle: Advanced
+layout: main.pug
+category: Advanced
+withHeadings: true
+navWeight: 4
+editable: true
+githubSource: docs/src/advanced/custom-backend.md
+---
+
+> This guide is compatible with Vue InstantSearch ≥ 1.6.0.
+
+Advanced InstantSearch users may have the need to query Algolia’s servers from their backend instead of the frontend, while still being able to reuse InstantSearch widgets. Possible motivations could be for security restrictions, for SEO purposes or to enrich the data sent by the custom server (i.e. fetch Algolia data and data from their own servers). If this sounds appealing to you, feel free to follow this guide. Keep in mind though that we, at Algolia, [recommend doing frontend search](https://www.algolia.com/doc/faq/searching/searching-from-the-front-end-or-the-back-end/#we-definitely-recommend-frontend-search) for performance and high availability reasons.
+
+By the end of this guide, you will have learned how to leverage InstantSearch with your own backend architecture to query Algolia.
+
+## How it’s made possible
+
+InstantSearch is the UI part on top a search client with a state managed by a [JavaScript Helper](https://github.com/algolia/algoliasearch-helper-js). These three layers are composable and can be interchanged to leverage the InstantSearch widgets system with different search clients.
+
+The [search client](https://github.com/algolia/algoliasearch-client-javascript) that Algolia offers queries Algolia’s backends whenever the user refines the search. It is possible to implement your own search client that queries your backend, which then queries Algolia’s backends with the [Algolia search client](https://github.com/algolia/algoliasearch-client-javascript) on your server.
+
+To create your own client, you will need to implement a given interface that receives and returns formatted data that InstantSearch can understand.
+
+## On the backend: create the necessary routes
+
+This guide assumes that you’ve got an existing server running on [http://localhost:3000](http://localhost:3000) with the route `POST /search` that takes the default Algolia query parameters as JSON. This backend could be using the [JavasScript API client](https://www.algolia.com/doc/api-client/javascript/getting-started/) to query Algolia, on top of any other operations you want to perform.
+
+The `algoliasearch` package will allow you to query Algolia from your backend. Here's an example using [Express](https://expressjs.com/):
+
+```javascript
+// Instantiate an Algolia client
+const algoliasearch = require('algoliasearch');
+const algoliaClient = algoliasearch('YOUR_APP_ID', 'YOUR_SEARCH_API_KEY');
+
+// Add the search endpoint
+app.post('/search', async (req, res) => {
+  const { requests } = req.body;
+  const results = await algoliaClient.search(requests);
+  res.status(200).send(results);
+});
+```
+
+### Supporting *Search For Facet Values*
```

Comment:
Vue InstantSearch (v1) doesn't have SFFV, so this section can be removed until v2